include/libtorrent/alert_types.hpp

```diff
@@ -975,16 +975,23 @@ namespace libtorrent
 	{
 		// internal
 		save_resume_data_alert(aux::stack_allocator& alloc
-			, std::shared_ptr<entry> const& rd
+			, add_torrent_params params
 			, torrent_handle const& h);
 
 		TORRENT_DEFINE_ALERT_PRIO(save_resume_data_alert, 37)
 
 		static const int static_category = alert::storage_notification;
 		virtual std::string message() const override;
 
+		// the ``params`` structure is populated with the fields to be passed to
+		// add_torrent() or async_add_torrent() to resume the torrent. To
+		// save the state to disk, you may pass it on to write_resume_data().
+		add_torrent_params params;
```

Comment:
it could be const, but I've avoided it in other alerts since, technically, it may be beneficial for clients to *move* values out of alerts, and that may apply here too